micrometer-test/src/main/java/io/micrometer/core/tck/MeterRegistryCompatibilityKit.java

```diff
@@ -634,14 +635,23 @@ void recordWithSample() {
         @Test
         @DisplayName("record with stateful Observation instance")
         void recordWithObservation() {
-            Observation observation = Observation.start("myObservation", observationRegistry);
-            clock(registry).add(10, TimeUnit.NANOSECONDS);
+            Observation observation = Observation.createNotStarted("myObservation", observationRegistry)
+                    .lowCardinalityKeyValue("staticTag", "42").start();
+
+            // created after start, LongTaskTimer won't have it
+            observation.lowCardinalityKeyValue("dynamicTag", "24");
```

Comment:
In general that shouldn't really happen. We will recommend to the users that they create a non started observation, and always use a key value provider (I think that in extremely rare cases they could set key value pairs manually)

Reply:
The `KeyValueProvider` is the same case since it is queried at `stop` when you have all the details in the context and the `LongTaskTimer` is created at `start`.